bundles/org.openhab.binding.mystrom/src/main/java/org/openhab/binding/mystrom/internal/MyStromHandler.java

```diff
@@ -115,37 +104,19 @@ private void pollDevice() {
         }
     }
 
-    @Override
-    public void initialize() {
-        MyStromConfiguration config = getConfigAs(MyStromConfiguration.class);
-        this.hostname = HTTP_REQUEST_URL_PREFIX + config.hostname;
-
-        updateStatus(ThingStatus.UNKNOWN);
-        pollingJob = scheduler.scheduleWithFixedDelay(this::pollDevice, 0, config.refresh, TimeUnit.SECONDS);
-    }
-
-    @Override
-    public void dispose() {
-        if (pollingJob != null) {
-            pollingJob.cancel(true);
-            pollingJob = null;
-        }
-        super.dispose();
-    }
-
     /**
      * Given a URL and a set parameters, send a HTTP GET request to the URL location
      * created by the URL and parameters.
      *
-     * @param url The URL to send a GET request to.
+     * @param action The action we want to take
      * @return String contents of the response for the GET request.
-     * @throws Exception
+     * @throws MyStromException Throws on communication error
      */
     public String sendHttpGet(String action) throws MyStromException {
```

Comment:
Did you think about moving all these methods (`sendHttpGet` and `sendHttpPost` and one more I can't remember) to the abstract handler? If another device is added, it'll likely use one of these and the code can be shared.

Reply:
Good. I wasn't sure about that either. In the original addon there was only one Thing and the method was already called sendHttpGet. But the GET requests that support the plug are only supported by the plug, for example. On the other hand, the Plug does not support POST requests from the Bulb. So these methods are device-specific. Perhaps the name is inconveniently chosen and they should be private.

Reply:
Ah, with the getDeviceSpecificInformation method below, that does make sense.